### DISCUSSION TITLE
Supported? Beautiful 360° endless potentiometer! Alpha RV112FF-40B1

### DISCUSSION THREAD

**Discussion by stllvdv** [shanteacontrols/OpenDeck#286] [Q&A]
Hi @paradajz et all,

Been ordering parts A LOT lately, mainly pots and encoders, some switches… 

Found the perfect potentiometer, but I’ve never been super happy with encoders a) how most encoders feel b) how encoders behave musically. Ableton Push 3 and – surprisingly — the pretty inexpensive AKAI MPD218 changed how I think about "encoders", those just FEEL and behave GREAT, like great analog pots, technically being "360° endless rotary potentiometers"? I think the Alpha RV112FF-40B1 is used in a couple of devices by Native Instruments and e.g. Novation as well.

4 pins basically… Would it take a lot of work for these to be supported in Open Deck? They feel sooooo good to turn and make music with! Right now values start rising from 0 to 127 and change direction upon more rotation.

These would be a true game-changer to me:

https://www.mouser.de/ProductDetail/317-1240FF-15F-0B20K

https://www.mouser.de/datasheet/2/13/RV112FF-2303564.pdf

Found this project:

https://hackaday.io/project/171841-driveralgorithm-for-360-deg-endless-potentiometer

and this:

https://gerotakke.de/ottopot/

Warm regards!

**Comment by paradajz** [comment]
They do indeed look interesting. 

> Would it take a lot of work for these to be supported in Open Deck?

Most likely yes. As I've mentioned in other discussions, I am no longer implementing complex features without financial contribution.

**Reply by stllvdv** [reply]
They feel absolutely fantastic to make music with, I wouldn't touch another encoder in my life if I'd be able to implement these in a custom-build. Financial contributions: let me know what you have in mind.

**Comment by rolfdegen** [comment]
Here's my idea to make the Alpha potentiometers a little smoother :)

DIY Synth & Sampler https://forum.pjrc.com/index.php?threads/new-polyhonic-teensy-diy-sampler.76219/post-359206

**Reply by stllvdv** [reply]
@rolfdegen That's a cool idea for making these smoother. Though I actually LOVE the feel and definitely so much more than ANY other encoder which is why I'd love to use these. Are you able to make these 4 pin work with Open Deck?

**Comment by Wapata** [comment]
Hi.
In the end, is it not a "simple" pot like others ? One of the leg seem to be only offset.
<img width="729" height="214" alt="image" src="https://github.com/user-attachments/assets/85f9e918-65af-4f4c-b036-d4b95427b348" />
But it seem to infinitly scroll from 0 to 100% resistance by 180° travel
<img width="377" height="341" alt="image" src="https://github.com/user-attachments/assets/e58da1c2-2fef-4c39-af27-5c81fe85ec6e" />
(and that explain the fourth leg).

Are you sure you don't just need an encoder without "clicky rotation", smooth ?

**Reply by stllvdv** [reply]
@Wapata It's about the feel of these, they feel better than all the other encoders I demoed/purchased.

**Comment by rolfdegen** [comment]
Hallo..
The endless pots require two analog inputs. Unlike encoders, the endless pots don't bounce. The maximum resolution depends on the AD converter. Here's an example code.

#include <ResponsiveAnalogRead.h>

const int ANALOG_PIN1 = A0;
const int ANALOG_PIN2 = A1;

analog.update();  // read channel 1
analog1.update(); //  read channel 2

        if (analog.hasChanged() && analog1.hasChanged())
        {
            int value_A = analog.getValue();  // value_A  0 - 1022
            int value_B = analog1.getValue(); // value_B  0 - 1022

            Serial.print(value_A);
            Serial.print("  ");
            Serial.print(value_B);
            Serial.print("  ");

            // Compute current angle and dial position
            double fx = ((double)value_A / 511.5) - 1; // range -1 to +1
            double fy = ((double)value_B / 511.5) - 1;
            double angle = atan2(fx, fy); // range -pi to +pi
            Serial.print(angle);
            Serial.print("  ");
            int StartPos = 0;
            int EndPos = 127;
            int dialPosition = map(angle, -PI, PI, StartPos, EndPos);
            Serial.println(dialPosition);
        }

**Comment by Wapata** [comment]
Thanks for the details, it was interesting :-)
